1/ToDoApp/src/components/ToDoItem/ToDoItem.spec.js

```diff
@@ -0,0 +1,46 @@
+import { mount } from '@vue/test-utils'
+import ToDoItem from './ToDoItem.vue'
+
+describe('ToDoItem', () => {
+  describe('given a todo', () => {
+    const wrapper = mount(ToDoItem, {
+      propsData: {
+        todo:{
+          text: 'test-todo'
+        }
+      }
+    })
+    it('renders todo.text', () => {
+      expect(wrapper.vm.todo.text).toBe('test-todo');
+    })
+    it('does not show input field', () => {
+      expect(wrapper.vm.editModeOn).toBe(false);
+    })
+    describe('click on Edit Button', () => {
+      const editButton=wrapper.find('#editButton');
+      it('shows input field', () => {
+        expect(wrapper.find('#editButton').exists()).toBe(true);
+        editButton.trigger('click');
+        expect(wrapper.vm.editModeOn).toBe(true);
+      })
+      describe('edit text and submit', () => {
+        it('$emits save with edited todo', () => {
+          const updatedTodoText='updated todo';
+          wrapper.vm.$emit('editTodo', updatedTodoText);
+          expect(wrapper.emitted().editTodo[0]).toEqual([updatedTodoText]);
+          wrapper.vm.save();
+          expect(wrapper.vm.editModeOn).toBe(false);
```

Comment:
Aim for one expectation per test case. It's nothing bad to have a little repetition in your test cases. If you want to DRY your code, use `beforeEach`.